apps/project-renderer/src/components/app/index.js

```diff
@@ -98,8 +98,9 @@ const App = ( {
 					}
 
 					setContent( json.content );
-					setHasResponded( true );
-					// all the setters should be called here: page, responseHash, content and startTime
+					setHasResponded( json.done );
+					setResponseHash( json.r );
+					setCurrentPage( parseInt( json.p ) );
```

Comment:
Make sure to add the base to `parseInt` -> `parseInt( json.p, 10 )` [reference](https://stackoverflow.com/questions/6611824/why-do-we-need-to-use-radix-parameter-when-calling-parseint).
Also, is there any change we can just have it be a number in the first place?

Reply:
The cryptic response props are there to honor our legacy responses' data structure. The `p` stands for "page" and it's sent as an `int`, but I though we'd just make sure of it on the frontend.